src/main/kotlin/three/consulting/epoc/service/CustomerService.kt

```diff
@@ -33,6 +34,14 @@ class CustomerService(private val customerRepository: CustomerRepository) {
         else throw UnableToUpdateCustomerException()
     }
 
+    fun deleteCustomer(customerId: Long) {
+        try {
+            customerRepository.deleteById(customerId)
+        } catch (e: EmptyResultDataAccessException) {
+            throw UnableToDeleteCustomerException(customerId, e)
+        }
+    }
 }
 class UnableToCreateCustomerException : RuntimeException("Cannot create a customer with existing id")
 class UnableToUpdateCustomerException : RuntimeException("Cannot update customer, missing customer id")
+class UnableToDeleteCustomerException(id: Long, exception: EmptyResultDataAccessException) : RuntimeException("Cannot delete customer, no customer found for the given id: $id", exception)
```

Comment:
Ok, but very long lines like this you should split. Here a newline tight after the `:` would be along ktlint guidelines